polaris/dataset/_competition_dataset.py

```diff
@@ -0,0 +1,42 @@
+from typing import Optional
+
+from pydantic import model_validator
+from polaris.dataset import Dataset
+from polaris.utils.errors import PolarisChecksumError
+
+_CACHE_SUBDIR = "datasets"
+
+
+class CompetitionDataset(Dataset):
```

Comment:
True, good idea. This could be a reason to keep the `CompetitionDataset` class even though we don't need the `masked_md5sum` attribute anymore. I would prefer to add this and keep it.